design/re-introspection-interface.md

```diff
@@ -0,0 +1,88 @@
+# re-inspection API
+
+## Status
+
+implementable
+
+## Summary
+
+A declarative API is proposed to request the baremetal operator to
+inspect a `Ready` BareMetalHost.
+
+## Motivation
+
+We would like to have an interface to allow a user to re-gather hardware
+inventory of a `Ready` BareMetalHost when a hardware replacement is made.
+When a user of the underlying infrastructure makes some changes to the actual
+server (e.g. replace or add NIC, disk, etc.), the latest hardware inventory
+including those changes need to be re-collected and updated on the spec of the
+corresponding BareMetalHost object without having to delete it.
+
+Implementation of this proposal is based on using annotation (similar to
+[Reboot API](https://github.com/metal3-io/metal3-docs/blob/master/design/reboot-interface.md))
+to request inspection of a `Ready` BareMetalHost.
+Once the annotation is set on BareMetalHost, the baremetal operator will
+request hardware inspection of the host from Ironic.
+
+### Goals
+
+- A declarative API to perform inspection
+- Use this API for future Metal³ remediation controller
+
+### Non-Goals
+
+- Automated mechanism to trigger inspection
+
+## Proposal
+
+We follow Reboot API implementation and expect to implement the similar
+annotation based interface for a user to trigger inspection.
+
+`inspection.metal3.io` annotation form on BareMetalHost object
```

Comment:
I slightly prefer `inspect` (imperative verb) to `inspection` (noun) for consistency with reboot, but I'll let others comment.

Reply:
I agree with you. It was my bad with copy-paste.Updated.